### PR TITLE
MAINT: Updates meson docs and remove `mesondev.sh`

### PR DESCRIPTION
**Pull request by Smit-create**
See: https://github.com/rgommers/scipy/pull/94#issuecomment-984539223

cc @rgommers

### CODE REVIEW

File: MESON_BUILD.md [hunk 1]
```diff
@@ -98,12 +98,41 @@ working with the default `distutils`-based build one gets with
...
+To run the tests for a particular submodule(let's say `optimize`), you can use:
+
+```bash
+python dev.py -n -s optimize
+```
+
+#### You can also build, install and test `scipy` using the single command:
+
+```bash
+python dev.py -n
```
Comment (rgommers) [inline MESON_BUILD.md:133]: This should not use `-n` right?

**Review by rgommers — COMMENTED** [review]
LGTM, thanks @Smit-create. There's still one use of `mesondev.sh` left in Quickstart in the docs.

File: MESON_BUILD.md (hunk 1 above)
Comment (Smit-create) [reply MESON_BUILD.md:133]: Oops! you are right.

**Review by rgommers — APPROVED** [review]
LGTM now, thanks Smit!

**Pull request merged**